### PR TITLE
add rsuid and rnosymfollow

### PR DESCRIPTION
**Pull request by lengrongfu**
issue: https://github.com/containers/youki/issues/1534

- [x] rsuid
- [x] rnosymfollow

### CODE REVIEW

**Comment by YJDoc2** [comment]
Hey @lengrongfu , Thanks a lot for this! I haven't taken a detailed look, but I had a general question :

You have worked a lot with this part, and implemented several of these mount attr tests. Do you think we can refactor the test logic? What I mean is that is there a common "pattern" in all mount attr tests, that can be extracted in a function and then each individual attr test call this function ; so that individual attr test functions become simpler?

And this is by no means a request to do it in this PR. If that can be done, we might first implement all the tests and then do that in a separate PR. I am just curious to know if that might be possible or not.

Thanks :)

File: tests/rust-integration-tests/runtimetest/src/utils.rs [hunk 1]
```diff
@@ -401,3 +403,51 @@ pub fn test_mount_rstrictatime_option(path: &str) -> Result<(), std::io::Error>
     }
     Ok(())
 }
+
+pub fn test_mount_rnosymfollow_option(path: &str) -> Result<(), std::io::Error> {
+    let path = format!("{}/{}", path, "link");
+    // 获取文件元数据
```
Comment (utam0k) [inline tests/rust-integration-tests/runtimetest/src/utils.rs:409]: Please translate to English.

File: tests/rust-integration-tests/integration_test/src/tests/mounts_recursive/mod.rs [hunk 2]
```diff
@@ -202,6 +205,40 @@ fn check_recursive_nosuid() -> TestResult {
...
+        .set_destination(mount_dest_path)
+        .set_typ(None)
+        .set_source(Some(rsuid_dir_path.clone()))
+        .set_options(Some(mount_options));
+    let spec = get_spec(
+        vec![mount_spec],
+        vec!["runtimetest".to_string(), "mounts_recursive".to_string()],
+    );
+    let result = test_inside_container(spec, &|_| {
+        // touch /tmp/rsuid_dir/file
```
Comment (utam0k) [inline tests/rust-integration-tests/integration_test/src/tests/mounts_recursive/mod.rs:225]: I think this comment is unnecessary as this program is obvious.

File: tests/rust-integration-tests/integration_test/src/tests/mounts_recursive/mod.rs [hunk 3]
```diff
@@ -202,6 +205,40 @@ fn check_recursive_nosuid() -> TestResult {
...
+        let original_file_path = format!("{}/{}", rsuid_dir_path.to_str().unwrap(), "file");
+        File::create(original_file_path.clone())?;
+        // chmod +s /tmp/rsuid_dir/file && chmod +g /tmp/rsuid_dir/file
```
Comment (utam0k) [inline tests/rust-integration-tests/integration_test/src/tests/mounts_recursive/mod.rs:228]: Don't `nix` or `libc` crates have the interface of chmod?

File: tests/rust-integration-tests/runtimetest/src/utils.rs [hunk 4]
```diff
@@ -401,3 +403,49 @@ pub fn test_mount_rstrictatime_option(path: &str) -> Result<(), std::io::Error>
...
+                format!("get file symlink_metadata err {path:?}, {e}"),
+            ));
+        }
+    };
+    // check symbolic is followed
+    if metadata.file_type().is_symlink() && metadata.mode() & 0o777 == 0o777 {
+        println!("Symbolic link cannot be followed.");
+        Ok(())
+    } else {
+        println!("Symbolic link can be followed.");
```
Comment (utam0k) [inline tests/rust-integration-tests/runtimetest/src/utils.rs:423]:
```suggestion
        println!("symbolic link cannot be followed.");
        Ok(())
    } else {
        println!("symbolic link can be followed.");
```

File: tests/rust-integration-tests/integration_test/src/tests/mounts_recursive/mod.rs [hunk 5]
```diff
@@ -492,6 +526,45 @@ fn check_recursive_rstrictatime() -> TestResult {
...
+    );
+    let result = test_inside_container(spec, &|_| {
+        let original_file_path = format!("{}/{}", rnosymfollow_dir_path.to_str().unwrap(), "file");
+        File::create(original_file_path.clone())?;
+        let link_file_path = format!("{}/{}", rnosymfollow_dir_path.to_str().unwrap(), "link");
+        println!("original file: {original_file_path:?},link file: {link_file_path:?}");
+        let mode = libc::S_ISUID | libc::S_ISGID;
+        let result = unsafe { libc::chmod(original_file_path.as_ptr() as *const i8, mode) };
+        if result == -1 {
+            eprintln!("Error: {:?}", std::io::Error::last_os_error());
```
Comment (utam0k) [inline tests/rust-integration-tests/integration_test/src/tests/mounts_recursive/mod.rs:557]: Don't we need to return here?

**Review by utam0k — APPROVED** [review]
@lengrongfu Thanks!

**Pull request merged**